cypress/integration/Homework/calculator.spec.js

```diff
@@ -0,0 +1,203 @@
+
+            beforeEach("Selects the 'Integers only' marker based on the integersOnly value", () => {
+
+                clearBothInputFields()
+
+                if (numberArray.integersOnly) {
+                    calculator.integerSelection().click()
+                }
+            })
+
+            it(formItSentence("3", "1", "Add", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Add')
+                clearAndCheckAnswerField(false)
+            })
+
+            it(formItSentence("3", "2", "Subtract", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Subtract')
+                clearAndCheckAnswerField(false)
+            })
+
+            it(formItSentence("3", "3", "Multiply", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Multiply')
+                clearAndCheckAnswerField(false)
+            })
+
+            it(formItSentence("3", "4", "Divide", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field or gets the '" + divideByZeroError + "' error"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Divide')
+                checkDivisionAnswerOrError(numberArray, false, buildVersion, divideByZeroError, true)
+            })
+
+            it(formItSentence("3", "5", "Concatenate", "numbers", numberArray.firstNumber, numberArray.secondNumber, numberArray.integersOnly, "and clears the answer field"), () => {
+                performCalculatorOperation(numberArray.firstNumber, numberArray.secondNumber, 'Concatenate')
+                clearAndCheckAnswerField(true)
+            })
+        })
+    })
+})
+
+function visitPageSelectBuildIntegersOnly(integersOnly, integerValidation, buildVersion) {
+    calculator.visitInitialPage()
+    calculator.buildDropDownList().select(buildVersion)
+
+    if (integersOnly && integerValidation) {
+        calculator.integerSelection().click()
+    }
+}
+
+function beforeSectionForPageAndIntegers(integersOnly, integerValidation, buildVersion) {
+    before("Selects the 'Integers only' marker based on the integersOnly value and visits the initial page", () => {
+        visitPageSelectBuildIntegersOnly(integersOnly, integerValidation, buildVersion)
+    })
+}
+
+function clearBothInputFields() {
+    calculator.firstNumberField().clear()
+    calculator.secondNumberField().clear()
+}
+
+function beforeEachClearBothInputFields() {
```

Comment:
I would get rid of this function at all. Simple beforeEach above the tests calling clear function would be enough as it would be easier to add more items in it.